core/src/main/java/io/substrait/expression/EnumArg.java

```diff
@@ -0,0 +1,24 @@
+package io.substrait.expression;
+
+import io.substrait.function.SimpleExtension;
+import org.immutables.value.Value;
+
+@Value.Immutable
+/**
+ * Capture the enum value and the associated `SimpleExtension.EnumArgument` TODO: probably better to
+ * capture `SimpleExtension.Function` also.
+ */
+public interface EnumArg extends FunctionArg {
```

Comment:
Would `EnumLiteral` be a better name and wouldn't it be simpler to have `EnumArg extends Expression`? Then you wouldn't need `exprArguments()` above and you could add something like this to `ExpressionVisitor`:
```
  R visit(Expression.EnumLiteral expr) throws E;
```

Reply:
- Didn't make them Expressions because Enums don't have an associated `Type`. But can  extend Expression and model them as `StringType`. Do you anticipate any issues with this?
- Currently Enum's are tied to a `Function`; that is why I called them `EnumArg`. I was going to investigate moving EnumArg inside `Function`.
- I was wondering about the thought process here.. Alternate could be having an `EnumType` and then Functions could have `args` of the `EnumType`.

Reply:
I like the idea of Enum extending Expression and having an EnumType.